satellite/metabase/rangedloop/rangedlooptest/countobserver.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2022 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package rangedlooptest
+
+import (
+	"context"
+	"time"
+
+	"storj.io/storj/satellite/metabase/rangedloop"
+	"storj.io/storj/satellite/metabase/segmentloop"
+)
+
+// CountObserver is a subscriber to the segment loop which counts the number of segments.
+type CountObserver struct {
+	NumSegments int
+}
+
+// Start is the callback for segment loop start.
+func (c *CountObserver) Start(ctx context.Context, time time.Time) error {
+	c.NumSegments = 0
+	return nil
+}
+
+// Fork splits the observer to count ranges of segments.
+func (c *CountObserver) Fork(ctx context.Context) (rangedloop.Partial, error) {
+	// return new instance for threadsafety
+	return &CountObserver{}, nil
+}
+
+// Join adds the count of all the ranges together.
+func (c *CountObserver) Join(ctx context.Context, partial rangedloop.Partial) error {
+	countPartial := partial.(*CountObserver)
+	c.NumSegments += countPartial.NumSegments
+	// Range done
+	return nil
+}
+
+// Finish is the callback for segment loop end.
```

Comment:
ranged loop end?

Reply:
Changed to "ranged segment loop"

That it processes segments is more important than that it works with ranges.